cc_core.c

```diff
@@ -3664,13 +3665,12 @@ int global_static_array(struct type* type_size, char* name)
 	require_match("missing close bracket\n", "]");
 	require_match("missing ;\n", ";");
 
-	globals_list = emit("\n'", globals_list);
-	while (0 != size)
+	unsigned i = ceil_div(size, register_size);
+	while(i != 0)
 	{
-		globals_list = emit(" 00", globals_list);
-		size = size - 1;
+		globals_list = emit("NULL\n", globals_list);
+		i = i - 1;
```

Comment:
This will make char take more space in the binary, woudn't it? Though I guess it's fine as long as the code is simpler...

Reply:
As far as I can tell `NULL` is defined in the `*_defs.M1` files to be the same as `register_size`. So this shouldn't take up more space in the binary it should just be a simpler way of outputting it.

Reply:
Yeah, OK. As long as we stick to the convention that NULL is register sized we should be fine.